src/components/Tree/useFlattenTreeData/useFlattenTreeData.js

```diff
@@ -128,13 +128,18 @@ const useFlattenTreeData = ({
   )
 
   const handleSetSelectedNodesFromKeysArr = useCallback(
-    selectedKeysArr => {
+    (keysToAdd, keysToRemove = []) => {
       if (!Object.keys(flattenedNodes).length) return
-      selectedKeysArr.forEach(
+      keysToRemove.forEach(
+        key =>
+          (flattenedNodes[key] = { ...flattenedNodes[key], isSelected: false }),
+      )
```

Comment:
Here you could use `reduce`,
To keep `flattenedNodes` immutable.
(If you want)
Like this:
```suggestion
    const newFlattenedNodes = keysToRemove.reduce((newFlattenedNodes, key) => (
      { ...newFlattenedNodes[key], isSelected: false }
    ), flattenedNodes)
```

Reply:
Many functions are mutating the global state. So for this to work the whole thing will need to be refactored. So never mind for now.